includes/class-wc-logger.php

```diff
@@ -5,157 +5,195 @@
 }
 
 /**
- * Allows log files to be written to for debugging purposes
+ * Provides logging capabilities for debugging purposes.
  *
  * @class          WC_Logger
- * @version        1.6.4
+ * @version        2.0.0
  * @package        WooCommerce/Classes
  * @category       Class
  * @author         WooThemes
  */
 class WC_Logger {
 
 	/**
-	 * Stores open file _handles.
+	 * Log Levels
+	 *
+	 * @see @link {https://tools.ietf.org/html/rfc5424}
+	 */
+	const DEBUG     = 'debug';
+	const INFO      = 'info';
+	const NOTICE    = 'notice';
+	const WARNING   = 'warning';
+	const ERROR     = 'error';
+	const CRITICAL  = 'critical';
+	const ALERT     = 'alert';
+	const EMERGENCY = 'emergency';
+
+	/**
+	 * Stores registered log handlers.
 	 *
 	 * @var array
 	 * @access private
 	 */
-	private $_handles;
+	private $handlers;
+
+	private static $valid_levels = array(
+		self::DEBUG,
+		self::INFO,
+		self::NOTICE,
+		self::WARNING,
+		self::ERROR,
+		self::CRITICAL,
+		self::ALERT,
+		self::EMERGENCY,
+	);
 
 	/**
 	 * Constructor for the logger.
 	 */
 	public function __construct() {
-		$this->_handles = array();
+		$handlers = apply_filters( 'woocommerce_register_log_handlers', array() );
+		$this->handlers = $handlers;
 	}
 
 	/**
-	 * Destructor.
+	 * Add a log entry.
+	 *
+	 * @deprecated since 2.0.0
+	 *
+	 * @param string $handle
+	 * @param string $message
+	 *
+	 * @return bool
 	 */
-	public function __destruct() {
-		foreach ( $this->_handles as $handle ) {
-			if ( is_resource( $handle ) ) {
-				fclose( $handle );
-			}
-		}
+	public function add( $handle, $message ) {
+		_deprecated_function( 'WC_Logger::add', '2.8', 'WC_Logger::log' );
+		$message = apply_filters( 'woocommerce_logger_add_message', $message, $handle );
+		$this->log( self::INFO, $message, array( 'tag' => $handle, '_legacy' => true ) );
```

Comment:
`INFO` seems the most neutral to me. Anything higher seems to indicate a problem of some sort, and `DEBUG`, the only lower level, seems to indicate the message is trivial. This is just a feeling, I do not have any technical reasons to chose `INFO` or any other level as the default for `add`. 

For reference, here are the [descriptions](https://tools.ietf.org/html/rfc5424):

```
              0       Emergency: system is unusable
              1       Alert: action must be taken immediately
              2       Critical: critical conditions
              3       Error: error conditions
              4       Warning: warning conditions
              5       Notice: normal but significant condition
              6       Informational: informational messages
              7       Debug: debug-level messages
```